### PR TITLE
compilable under BSD

### PR DESCRIPTION
**Pull request by devnexen**
This branch is to make it possible to compile under BSD (tested under Free and OpenBSD), avoiding linuxisms. Hope it finds its way :)

### CODE REVIEW

**Comment by x42** [comment]
Odd that `obj.linkflags = bld.env['LDFLAGS']` is needed. waf adds that automatically.

I just tried   `LDFLAGS="-L/testing" ./waf configure`
it's printed in the config log and also used when linking (check with ./waf -v)

File: libs/ardour/wscript [hunk 1]
```diff
@@ -373,7 +373,7 @@ def build(bld):
                         'SNDFILE','SAMPLERATE','LRDF','AUDIOUNITS', 'GIOMM',
                         'OSX','BOOST','CURL','TAGLIB','VAMPSDK','VAMPHOSTSDK','RUBBERBAND']
     obj.use          = ['libpbd','libmidipp','libevoral',
-                        'libaudiographer',
+                        'libaudiographer', 'libexecinfo',
```
Comment (x42) [inline libs/ardour/wscript]: `libexecinfo` should be probably be added as uselib "EXECINFO" or otherwise special cased for BSD.

**Comment by devnexen** [comment]
true especially EXECINFO was defined I forgot to remove the unneeded one. Thanks for the feedback.

**Comment by x42** [comment]
looking at the combined diff ( https://patch-diff.githubusercontent.com/raw/Ardour/ardour/pull/227.diff )
there are still a few `obj.linkflags = bld.env['LDFLAGS']`  additions. Are those needed?

The ones to initialize it are fine, IMHO (allows to consistently use `+=` elsewhere) any thoughts, @pauldavisthefirst ?

File: gtk2_ardour/video_timeline.cc [hunk 2]
```diff
@@ -33,6 +33,10 @@
 #include "video_timeline.h"
 #include "video_tool_paths.h"
 
+#ifdef PLATFORM_BSD
+#include <sys/wait.h>
+#endif
+
```
Comment (x42) [inline gtk2_ardour/video_timeline.cc]: Why is this needed here?  `WNOHANG` ?  if so it might be better in `libs/pbd/pbd/system_exec.h`
where the `wait()` wrapper is defined.
